packages/client/tests/functional/metrics-disabled/tests.ts

```diff
@@ -0,0 +1,19 @@
+import testMatrix from './_matrix'
+import { expectTypeOf } from 'expect-type'
+
+// @ts-ignore this is just for type checks
+declare let prisma: import('@prisma/client').PrismaClient
+
+testMatrix.setupTestSuite(() => {
+  test('attempt to use $metrics a compile-time error', () => {
+    // TODO: this is always an error in the editor. Maybe we need to generate
```

Comment:
Do you mean that we need to have dev-time types/intellisense?

Reply:
Yea, basically, right now it always highlighted as an error in editors, since `prisma` has `any` type there and any can have anything. It works when you actually run the tests, since there `prisma` is properly typed by then. Having proper types during development will fix this. I'd maybe do it separately as a part of testing project